analyze/analyze.rs

```diff
@@ -402,7 +402,11 @@ impl traits::Emit for DominatorTree {
                 items.retained_size(id),
                 (items.retained_size(id) as f64) / (items.size() as f64) * 100.0,
             );
-
+            let idom = if let Some(idom) = items.predecessors(id).last() {
```

Comment:
This isn't quite right.

*Predecessors* are just the normal neighbors graph edges reversed. Instead of asking questions like "what other items does X refer to?" it allows us to ask "what items refer to X?"

The *immediate dominator* of X is the parent of item X in the dominators tree. Although we have it available, we don't actually save this information inside `ir::Items::compute_dominators` right now, so we will need to do that. We can add an `immediate_dominators: Option<BTreeMap<Id, Id>>` member to `ir::Items` and fill it in inside `compute_dominators`.

Does that make sense?

Reply:
Yes, fix later.